### ISSUE TITLE
Self-Update Support for Github apks

### ISSUE THREAD

**Issue by jerryn70** [LibreShift/red-moon#155]
For example, upon opening the app Red Moon it automatically checks for updates, and if found then it asks whether to ignore / update or remind later, after this there is no additional action required by the user. 

Take a look at this: https://github.com/javiersantos/AppUpdater

**Comment by smichel17** [comment]
I think that if Red Moon is installed through gplay or f-droid, we do not need to bother the user with update notifications. If they want those, they'll configure them through the respective app stores. If Red Moon is installed through its own apk download (eg, from github), this is a reasonable thing to do.

There is a library by the f-droid people for checking whether gplay/f-droid are installed: https://gitlab.com/fdroid/update-channels

**Comment by smichel17** [comment]
Thinking about this again, one downside is that we'd have to add internet permissions, which I'm a little hesitant to do given that we also ask for location and monitoring the active app. We only use location for calculating sunrise/sunset times and we only monitor the active app in order to pause in secure apps (or user-excluded apps). However, we *could* in a future update, and it's a lot to ask that users trust us with this.

So, I'm currently thinking of a middle ground: If we detect that f-droid is not installed (and red moon is not installed via gplay), we should periodically (every 6 months?) remind users of this (with a dialog when they open the app).

**Comment by smichel17** [comment]
I'm going to close this. The main reason is that I am preparing for one final push to get Red Moon feature-complete, and after this point, Red Moon will go into maintenance mode (bug fixes and support for new Android versions only). At this point, the only reason you would need to update is if you are affected by a bug, in which case the annoyance caused by the bug is as good a reminder to check for updates as a dialog would be.

Maintenance mode is not a bad thing: If an app is close enough to perfect that it cannot really be improved, there is no need to do more work on it. Just because it is not under development, does not mean people should stop using it. For example, [Talalarmo](https://github.com/trikita/talalarmo) is one such application. Apart from [one outstanding pull request (from me)](https://github.com/trikita/talalarmo/pull/26), it's basically perfect, even though it hasn't received a commit in 4 years. I still recommend it to people.

Red Moon is an application like this. The problem it solves is not *that* complicated;  I believe it is possible to get the app to a finished state, where it solves all the problems it is meant to solve, and the problems it does not solve would be better solved by a separate app.

**tl;dr it's not worth building update support for an app that's not going to receive many updates.**

Two other reasons:

- I personally was never the one distributing Red Moon. @raatmarien has done the Google Play and Github apks releases, but not for a while. So it's not worth building support for an update channel that won't be used.
- I think there's some value in having [separate maintainers and upstream developers](http://kmkeen.com/maintainers-matter/). So I am not sure I even want to offer direct apk downloads as a recommended install path. To some extent I'd rather point people to f-droid (or a 3rd party f-droid repository).
